### PR TITLE
TXModal: Extra content for results

### LINKED ISSUE #769: [TxModal] Customize content of result
by providing `extraResult`

### PR DESCRIPTION
**Pull request by veado**
- [x] Add `extraResult` props accepts any React.Node to add extra content for results. With it, we can show one or two "ViewTxButton" or anything else.
- [x] Extract `ViewTxButton` component (incl. stories)

Needed for #766 

 
Closes #769

### CODE REVIEW

File: src/renderer/components/modal/tx/TxModal.tsx [hunk 1]
```diff
@@ -56,41 +43,24 @@ export const TxModal: React.FC<Props> = (props): JSX.Element => {
...
+          disabled={RD.isInitial(txRD) || RD.isPending(txRD)}
+          color="success"
+          onClick={RD.isSuccess(txRD) ? onFinish : onClose}>
+          {intl.formatMessage({ id: RD.isFailure(txRD) ? 'common.cancel' : 'common.finish' })}
+        </Styled.ResultButton>
+        {renderExtraResult}
+      </Styled.ResultContainer>
     ),
-    [intl, oTxHash, onClose, onFinish, renderTxButton, txRD]
+    [intl, onClose, onFinish, renderExtraResult, txRD]
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/components/modal/tx/TxModal.tsx:63]: just IMHO it's better to keep things more explicit without any conditions at the "view"-layer. such approach provides clearer code for understanding and maintaining
```tsx
const renderResult = useMemo(() => {
    const resultButtonProps = FP.pipe(
      txRD,
      RD.fold(
        () => ({
          disabled: true,
          onClick: onClose,
          children: intl.formatMessage({ id: 'common.finish' })
        }),
        () => ({
          disabled: true,
          onClick: onClose,
          children: intl.formatMessage({ id: 'common.finish' })
        }),
        () => ({
          onClick: onClose,
          children: intl.formatMessage({ id: 'common.cancel' })
        }),
        () => ({
          onClick: onFinish,
          children: intl.formatMessage({ id: 'common.finish' })
        })
      )
    )

    return (
      <Styled.ResultContainer>
        <Styled.ResultButton color="success" {...resultButtonProps} />
        {renderExtraResult}
      </Styled.ResultContainer>
    )
  }, [intl, onClose, onFinish, renderExtraResult, txRD])
```

File: src/renderer/components/uielements/button/ViewTxButton.style.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+const ExternalLinkIcon = styled(UIExternalLinkIcon)`
+  svg {
+    color: ${palette('primary', 0)};
+  }
+`
+
+export const ViewTxButton = styled(UIButton).attrs({
+  typevalue: 'transparent',
+  icon: <ExternalLinkIcon />
+})``
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/components/uielements/button/ViewTxButton.style.tsx:18]: maybe we should handle disabled state for styles too ?

File: src/renderer/components/uielements/button/ViewTxButton.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+export const ViewTxButton: React.FC<Props> = ({ onClick, txHash: oTxHash, label }): JSX.Element => {
+  const intl = useIntl()
+
+  const onClickHandler = useCallback(
+    (_) => {
+      FP.pipe(
+        oTxHash,
+        O.fold(FP.constUndefined, (txHash) => onClick(txHash))
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/components/uielements/button/ViewTxButton.tsx:23]: looks like
```O.fold(FP.constUndefined, onClick)```
is enough

**Review by thatStrangeGuyThorchain — CHANGES_REQUESTED** [review]

File: src/renderer/components/modal/tx/TxModal.tsx [hunk 4]
```diff
@@ -1,40 +1,27 @@
...
-    txHash: oTxHash = O.none,
-    txRD,
-    startTime,
-    onClose,
-    onFinish,
-    onViewTxClick = FP.constVoid,
-    extra = <></>,
-    timerValue = NaN
-  } = props
+  const { title, txRD, startTime, onClose, onFinish, extra = <></>, extraResult, timerValue = NaN } = props
```
[the remark this answers is not in the document]
Comment (veado) [reply src/renderer/components/modal/tx/TxModal.tsx:24]: `TxModal` uses `TxTimer` internally, which started a custom counter if`startTime` is not set.

File: src/renderer/components/uielements/button/ViewTxButton.style.tsx (hunk 2 above)
Comment (veado) [reply src/renderer/components/uielements/button/ViewTxButton.style.tsx:18]: `UIButton` has already some styles for `disabled` state or do you want to add different styles, here?

**Pull request merged**